Mathlib/LinearAlgebra/TensorProduct.lean

```diff
@@ -216,12 +211,16 @@ theorem smul_tmul [DistribMulAction R' N] [CompatibleSMul R R' M N] (r : R') (m
   CompatibleSMul.smul_tmul _ _ _
 #align tensor_product.smul_tmul TensorProduct.smul_tmul
 
-attribute [local instance] addMonoid
+-- porting note: This is added as a local instance for `SMul.aux`.
+-- For some reason type-class inference in Lean 3 unfolded this definition.
+private def addMonoidWithWrongNSMul : AddMonoid (M ⊗[R] N) :=
+  { (addConGen (TensorProduct.Eqv R M N)).addMonoid with }
+
+attribute [local instance] addMonoidWithWrongNSMul in
```

Comment:
Inlining this into a `let` seems to cause more performance issues: #8582. I don't want to investigate that further in this PR, as I've made #8519 depend on the current PR.

Reply:
Why should #8519 depend on this PR? I think this PR is akin to documentation improvement rather than new feature that something else depends on?

Reply:
Because #8519 contains the same change